src/main/java/org/gridsuite/filter/server/FilterController.java

```diff
@@ -115,14 +103,13 @@ public FilterController(FilterService service) {
             .body(service.replaceFilterWithScript(id));
     }
 
-    @PostMapping(value = "/filters/{id}/new-script/{scriptId}/{scriptName}")
+    @PostMapping(value = "/filters/{id}/new-script")
     @Operation(summary = "Create a new script filter from a filter")
     @ApiResponses(value = {@ApiResponse(responseCode = "200", description = "The script filter have been created successfully")})
     public ResponseEntity<AbstractFilter> newScriptFromFilter(@PathVariable("id") UUID filterId,
-                                                              @PathVariable("scriptId") UUID scriptId,
-                                                              @PathVariable("scriptName") String scriptName) {
+                                                              @RequestParam(required = false, value = "newId") UUID newId) {
```

Comment:
I know it's not from your change but...
Why do we need to accept a `scriptId` or `newId` param ? Seems like it's not used in gridexplore-app... Maybe in the new explore-server ? But it's a risk of UUID duplication here no ? I mean we could pass a `newId` value which already exists in database no?
I continue to think we should generate UUID (`UUID.randomUUID()`) to avoid future problems of this type.